### ISSUE TITLE
Postgres question fails on 'Infinity' date value

### ISSUE THREAD

**Issue by alexpusch** [metabase/metabase#12761]
**Describe the bug**
When a postgres table includes the date "Infinity" fails with "There was a problem with your question" error. The network call itself seem to succeed and return some rows, but in middle of the result JSON there is an error message

**Logs**
Please include javascript console and server logs around the time this bug occurred. For information about how to get these, consult our [bug troubleshooting guide](https://metabase.com/docs/latest/troubleshooting-guide/bugs.html)

Network response: 

```
{"database_id":3,"started_at":"2020-06-23T12:22:01.923437Z","via":[{"status":"failed","class":"class clojure.lang.ExceptionInfo","error":"Error reducing result rows","stacktrace":["--> query_processor.context.default$default_reducef$fn__37885.invoke(default.clj:61)","query_processor.context.default$default_reducef.invokeStatic(default.clj:58)","query_processor.context.default$default_reducef.invoke(default.clj:49)","query_processor.context$reducef.invokeStatic(context.clj:69)","query_processor.context$reducef.invoke(context.clj:62)","query_processor.context.default$default_runf$respond_STAR___37889.invoke(default.clj:70)","driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:392)","driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:377)","driver.sql_jdbc$fn__72719.invokeStatic(sql_jdbc.clj:50)","driver.sql_jdbc$fn__72719.invoke(sql_jdbc.clj:48)","query_processor.context$executef.invokeStatic(context.clj:59)","query_processor.context$executef.invoke(context.clj:48)","query_processor.context.default$default_runf.invokeStatic(default.clj:69)","query_processor.context.default$default_runf.invoke(default.clj:67)","query_processor.context$runf.invokeStatic(context.clj:45)","query_processor.context$runf.invoke(context.clj:39)","query_processor.reducible$pivot.invokeStatic(reducible.clj:34)","query_processor.reducible$pivot.invoke(reducible.clj:31)","query_processor.middleware.mbql_to_native$mbql__GT_native$fn__43022.invoke(mbql_to_native.clj:26)","query_processor.middleware.check_features$check_features$fn__42336.invoke(check_features.clj:42)","query_processor.middleware.optimize_datetime_filters$optimize_datetime_filters$fn__43187.invoke(optimize_datetime_filters.clj:133)","query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__47085.invoke(wrap_value_literals.clj:137)","query_processor.middleware.annotate$add_column_info$fn__40965.invoke(annotate.clj:577)","query_processor.middleware.permissions$check_query_permissions$fn__42211.invoke(permissions.clj:64)","query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__43686.invoke(pre_alias_aggregations.clj:40)","query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__42409.invoke(cumulative_aggregations.clj:61)","query_processor.middleware.resolve_joins$resolve_joins$fn__44218.invoke(resolve_joins.clj:183)","query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__39152.invoke(add_implicit_joins.clj:245)","query_processor.middleware.limit$limit$fn__43008.invoke(limit.clj:38)","query_processor.middleware.format_rows$format_rows$fn__42989.invoke(format_rows.clj:81)","query_processor.middleware.desugar$desugar$fn__42475.invoke(desugar.clj:22)","query_processor.middleware.binning$update_binning_strategy$fn__41509.invoke(binning.clj:229)","query_processor.middleware.resolve_fields$resolve_fields$fn__42017.invoke(resolve_fields.clj:24)","query_processor.middleware.add_dimension_projections$add_remapping$fn__38688.invoke(add_dimension_projections.clj:272)","query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__38908.invoke(add_implicit_clauses.clj:147)","query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__39301.invoke(add_source_metadata.clj:105)","query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__43883.invoke(reconcile_breakout_and_order_by_bucketing.clj:98)","query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__41150.invoke(auto_bucket_datetimes.clj:125)","query_processor.middleware.resolve_source_table$resolve_source_tables$fn__42064.invoke(resolve_source_table.clj:46)","query_processor.middleware.parameters$substitute_parameters$fn__43668.invoke(parameters.clj:97)","query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__42116.invoke(resolve_referenced.clj:80)","query_processor.middleware.expand_macros$expand_macros$fn__42731.invoke(expand_macros.clj:158)","query_processor.middleware.add_timezone_info$add_timezone_info$fn__39332.invoke(add_timezone_info.clj:15)","query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__46969.invoke(splice_params_in_response.clj:32)","query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__43894$fn__43898.invoke(resolve_database_and_driver.clj:33)","driver$do_with_driver.invokeStatic(driver.clj:61)","driver$do_with_driver.invoke(driver.clj:57)","query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__43894.invoke(resolve_database_and_driver.clj:27)","query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__42937.invoke(fetch_source_query.clj:243)","query_processor.middleware.store$initialize_store$fn__46978$fn__46979.invoke(store.clj:11)","query_processor.store$do_with_store.invokeStatic(store.clj:46)","query_processor.store$do_with_store.invoke(store.clj:40)","query_processor.middleware.store$initialize_store$fn__46978.invoke(store.clj:10)","query_processor.middleware.cache$maybe_return_cached_results$fn__41993.invoke(cache.clj:208)","query_processor.middleware.validate$validate_query$fn__46987.invoke(validate.clj:10)","query_processor.middleware.normalize_query$normalize$fn__43035.invoke(normalize_query.clj:22)","query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__39170.invoke(add_rows_truncated.clj:36)","query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__46954.invoke(results_metadata.clj:128)","query_processor.middleware.constraints$add_default_userland_constraints$fn__42352.invoke(constraints.clj:42)","query_processor.middleware.process_userland_query$process_userland_query$fn__43757.invoke(process_userland_query.clj:136)","query_processor.middleware.catch_exceptions$catch_exceptions$fn__42295.invoke(catch_exceptions.clj:166)","query_processor.reducible$async_qp$qp_STAR___37971$thunk__37972.invoke(reducible.clj:100)","query_processor.reducible$async_qp$qp_STAR___37971.invoke(reducible.clj:106)","query_processor.reducible$sync_qp$qp_STAR___37980$fn__37983.invoke(reducible.clj:132)","query_processor.reducible$sync_qp$qp_STAR___37980.invoke(reducible.clj:131)","query_processor$process_userland_query.invokeStatic(query_processor.clj:218)","query_processor$process_userland_query.doInvoke(query_processor.clj:214)","query_processor$fn__47153$process_query_and_save_execution_BANG___47162$fn__47165.invoke(query_processor.clj:233)","query_processor$fn__47153$process_query_and_save_execution_BANG___47162.invoke(query_processor.clj:225)","query_processor$fn__47197$process_query_and_save_with_max_results_constraints_BANG___47206$fn__47209.invoke(query_processor.clj:245)","query_processor$fn__47197$process_query_and_save_with_max_results_constraints_BANG___47206.invoke(query_processor.clj:238)","api.dataset$fn__50495$fn__50498.invoke(dataset.clj:52)","query_processor.streaming$streaming_response_STAR_$fn__34009$fn__34010.invoke(streaming.clj:73)","query_processor.streaming$streaming_response_STAR_$fn__34009.invoke(streaming.clj:72)","async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:66)","async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:64)","async.streaming_response$do_f_async$fn__23079.invoke(streaming_response.clj:85)"],"error_type":"qp","ex-data":{"type":"qp"}}],"error_type":"qp","json_query":{"database":3,"type":"query","query":{"source-table":12},"parameters":[],"middleware":{"add-default-userland-constraints?":true}},"native":{"query":"SELECT \"public\".\"systemUpTime\".\"id\" AS \"id\", \"public\".\"systemUpTime\".\"endDate\" AS \"endDate\", \"public\".\"systemUpTime\".\"retailerId\" AS \"retailerId\", \"public\".\"systemUpTime\".\"startDate\" AS \"startDate\", \"public\".\"systemUpTime\".\"storeId\" AS \"storeId\" FROM \"public\".\"systemUpTime\" LIMIT 2000","params":null},"status":"failed","class":"class java.time.DateTimeException","stacktrace":["java.base/java.time.temporal.ValueRange.checkValidValue(Unknown Source)","java.base/java.time.temporal.ChronoField.checkValidValue(Unknown Source)","java.base/java.time.LocalDate.plusDays(Unknown Source)","java.base/java.time.LocalDateTime.plusWithOverflow(Unknown Source)","java.base/java.time.LocalDateTime.plusSeconds(Unknown Source)","java.base/java.time.OffsetDateTime.withOffsetSameInstant(Unknown Source)","java_time.zone$fn__4626.invokeStatic(zone.clj:340)","java_time.zone$fn__4626.invoke(zone.clj:331)","java_time.zone$fn__4609$G__4591__4616.invoke(zone.clj:314)","--> query_processor.middleware.format_rows$fn__42977.invokeStatic(format_rows.clj:56)","query_processor.middleware.format_rows$fn__42977.invoke(format_rows.clj:18)","query_processor.middleware.format_rows$fn__42944$fn__42947$G__42945__42954.invoke(format_rows.clj:12)","query_processor.middleware.format_rows$format_rows_xform$fn__42982$fn__42983.invoke(format_rows.clj:75)","query_processor.middleware.format_rows$format_rows_xform$fn__42982.invoke(format_rows.clj:75)","query_processor.middleware.limit$limit_xform$fn__43005.invoke(limit.clj:25)","query_processor.reducible$reducible_rows$reify__37987.reduce(reducible.clj:156)","query_processor.context.default$default_reducef$fn__37885.invoke(default.clj:59)","query_processor.context.default$default_reducef.invokeStatic(default.clj:58)","query_processor.context.default$default_reducef.invoke(default.clj:49)","query_processor.context$reducef.invokeStatic(context.clj:69)","query_processor.context$reducef.invoke(context.clj:62)","query_processor.context.default$default_runf$respond_STAR___37889.invoke(default.clj:70)","driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:392)","driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:377)","driver.sql_jdbc$fn__72719.invokeStatic(sql_jdbc.clj:50)","driver.sql_jdbc$fn__72719.invoke(sql_jdbc.clj:48)","query_processor.context$executef.invokeStatic(context.clj:59)","query_processor.context$executef.invoke(context.clj:48)","query_processor.context.default$default_runf.invokeStatic(default.clj:69)","query_processor.context.default$default_runf.invoke(default.clj:67)","query_processor.context$runf.invokeStatic(context.clj:45)","query_processor.context$runf.invoke(context.clj:39)","query_processor.reducible$pivot.invokeStatic(reducible.clj:34)","query_processor.reducible$pivot.invoke(reducible.clj:31)","query_processor.middleware.mbql_to_native$mbql__GT_native$fn__43022.invoke(mbql_to_native.clj:26)","query_processor.middleware.check_features$check_features$fn__42336.invoke(check_features.clj:42)","query_processor.middleware.optimize_datetime_filters$optimize_datetime_filters$fn__43187.invoke(optimize_datetime_filters.clj:133)","query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__47085.invoke(wrap_value_literals.clj:137)","query_processor.middleware.annotate$add_column_info$fn__40965.invoke(annotate.clj:577)","query_processor.middleware.permissions$check_query_permissions$fn__42211.invoke(permissions.clj:64)","query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__43686.invoke(pre_alias_aggregations.clj:40)","query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__42409.invoke(cumulative_aggregations.clj:61)","query_processor.middleware.resolve_joins$resolve_joins$fn__44218.invoke(resolve_joins.clj:183)","query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__39152.invoke(add_implicit_joins.clj:245)","query_processor.middleware.limit$limit$fn__43008.invoke(limit.clj:38)","query_processor.middleware.format_rows$format_rows$fn__42989.invoke(format_rows.clj:81)","query_processor.middleware.desugar$desugar$fn__42475.invoke(desugar.clj:22)","query_processor.middleware.binning$update_binning_strategy$fn__41509.invoke(binning.clj:229)","query_processor.middleware.resolve_fields$resolve_fields$fn__42017.invoke(resolve_fields.clj:24)","query_processor.middleware.add_dimension_projections$add_remapping$fn__38688.invoke(add_dimension_projections.clj:272)","query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__38908.invoke(add_implicit_clauses.clj:147)","query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__39301.invoke(add_source_metadata.clj:105)","query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__43883.invoke(reconcile_breakout_and_order_by_bucketing.clj:98)","query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__41150.invoke(auto_bucket_datetimes.clj:125)","query_processor.middleware.resolve_source_table$resolve_source_tables$fn__42064.invoke(resolve_source_table.clj:46)","query_processor.middleware.parameters$substitute_parameters$fn__43668.invoke(parameters.clj:97)","query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__42116.invoke(resolve_referenced.clj:80)","query_processor.middleware.expand_macros$expand_macros$fn__42731.invoke(expand_macros.clj:158)","query_processor.middleware.add_timezone_info$add_timezone_info$fn__39332.invoke(add_timezone_info.clj:15)","query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__46969.invoke(splice_params_in_response.clj:32)","query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__43894$fn__43898.invoke(resolve_database_and_driver.clj:33)","driver$do_with_driver.invokeStatic(driver.clj:61)","driver$do_with_driver.invoke(driver.clj:57)","query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__43894.invoke(resolve_database_and_driver.clj:27)","query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__42937.invoke(fetch_source_query.clj:243)","query_processor.middleware.store$initialize_store$fn__46978$fn__46979.invoke(store.clj:11)","query_processor.store$do_with_store.invokeStatic(store.clj:46)","query_processor.store$do_with_store.invoke(store.clj:40)","query_processor.middleware.store$initialize_store$fn__46978.invoke(store.clj:10)","query_processor.middleware.cache$maybe_return_cached_results$fn__41993.invoke(cache.clj:208)","query_processor.middleware.validate$validate_query$fn__46987.invoke(validate.clj:10)","query_processor.middleware.normalize_query$normalize$fn__43035.invoke(normalize_query.clj:22)","query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__39170.invoke(add_rows_truncated.clj:36)","query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__46954.invoke(results_metadata.clj:128)","query_processor.middleware.constraints$add_default_userland_constraints$fn__42352.invoke(constraints.clj:42)","query_processor.middleware.process_userland_query$process_userland_query$fn__43757.invoke(process_userland_query.clj:136)","query_processor.middleware.catch_exceptions$catch_exceptions$fn__42295.invoke(catch_exceptions.clj:166)","query_processor.reducible$async_qp$qp_STAR___37971$thunk__37972.invoke(reducible.clj:100)","query_processor.reducible$async_qp$qp_STAR___37971.invoke(reducible.clj:106)","query_processor.reducible$sync_qp$qp_STAR___37980$fn__37983.invoke(reducible.clj:132)","query_processor.reducible$sync_qp$qp_STAR___37980.invoke(reducible.clj:131)","query_processor$process_userland_query.invokeStatic(query_processor.clj:218)","query_processor$process_userland_query.doInvoke(query_processor.clj:214)","query_processor$fn__47153$process_query_and_save_execution_BANG___47162$fn__47165.invoke(query_processor.clj:233)","query_processor$fn__47153$process_query_and_save_execution_BANG___47162.invoke(query_processor.clj:225)","query_processor$fn__47197$process_query_and_save_with_max_results_constraints_BANG___47206$fn__47209.invoke(query_processor.clj:245)","query_processor$fn__47197$process_query_and_save_with_max_results_constraints_BANG___47206.invoke(query_processor.clj:238)","api.dataset$fn__50495$fn__50498.invoke(dataset.clj:52)","query_processor.streaming$streaming_response_STAR_$fn__34009$fn__34010.invoke(streaming.clj:73)","query_processor.streaming$streaming_response_STAR_$fn__34009.invoke(streaming.clj:72)","async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:66)","async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:64)","async.streaming_response$do_f_async$fn__23079.invoke(streaming_response.clj:85)"],"context":"ad-hoc","error":"Invalid value for Year (valid values -999999999 - 999999999): 1000000000","row_count":0,"running_time":0,"preprocessed":{"database":3,"type":"query","query":{"source-table":12,"fields":[["field-id",77],["datetime-field",["field-id",73],"default"],["field-id",75],["datetime-field",["field-id",76],"default"],["field-id",74]],"limit":2000},"middleware":{"add-default-userland-constraints?":true},"info":{"executed-by":2,"context":"ad-hoc","nested?":false,"query-hash":"0xF46AB51E"},"constraints":{"max-results":10000,"max-results-bare-rows":2000}},"data":{"rows":[],"cols":[]}}
```

Chrome dev tools logs:
```
app-main.bundle.js?1636e40125c6bad14763:5 Error getting setting table._cell_background_getter TypeError: Cannot read property 'rows' of undefined
    at Object.getValue (app-main.bundle.js?1636e40125c6bad14763:5)
    at g (app-main.bundle.js?1636e40125c6bad14763:5)
    at t.getComputedSettings (app-main.bundle.js?1636e40125c6bad14763:5)
    at h (app-main.bundle.js?1636e40125c6bad14763:5)
    at app-main.bundle.js?1636e40125c6bad14763:5
    at vendor.bundle.js?1636e40125c6bad14763:1
    at vendor.bundle.js?1636e40125c6bad14763:1
    at vendor.bundle.js?1636e40125c6bad14763:1
    at vendor.bundle.js?1636e40125c6bad14763:1
    at vendor.bundle.js?1636e40125c6bad14763:1
```

**To Reproduce**
1. Go to the view of a postgres table that contains a date value "infinity"
2. See error

**Expected behavior**
Metabase should display "infinity"

**Information about your Metabase Installation:**

- Your browser and the version: Chrome 83
- Your operating system: Ubuntu 18.04
- Your databases: Postgres
- Metabase version: 0.35.2
- Metabase hosting environment: Docker
- Metabase internal database: postgres

**Severity**
blocking use of certain tables

**Comment by flamber** [comment]
Hi @alexpusch 
There were some fixes to `Infinity` in 0.35.4 - #12511 - but can you try that version?

**Comment by sbelak** [comment]
Those were for numeric fields though. Until this moment I didn't knew that PG supports date infinities. The latest version will prevent the whole thing from crashing though, you just won't have fingerprints on those columns, but that shouldn't affect the usage too much.

**Comment by alexpusch** [comment]
Issue is not resolved in 0.35.4

**Comment by sbelak** [comment]
I see. It's a serialization problem then. Thank you, this is immensely helpful

**Comment by dpsutton** [comment]
```sql
create table times (stuff text, t timestamp);

insert into times (stuff, t) values ('hi', now()), ('infinity', 'infinity');

 select * from times;
  stuff   |             t
----------+----------------------------
 infinity | infinity
 hello    | 2020-10-19 16:23:28.711942
(2 rows)
```

<img width="435" alt="image" src="https://user-images.githubusercontent.com/6377293/96514694-9ce2cb00-1229-11eb-9ace-9705f526a529.png">

@alexpusch I'm not able to replicate. 
I inserted the above, then click "Ask a question", choose my pg db, select the table "Times" and I successfully get a preview. I'm running an as-yet unreleased 0.36.8 and wanted to see if you could replicate the bug with the above minimal example. Wondering whether this has gotten fixed in the meantime or if perhaps the example needs to be a bit richer to hit the bug.

**Comment by alexpusch** [comment]
Thanks for the response @dpsutton. With your example I don't get the error as well. It seems that the specific troublesome type is actually timestamptz and not timestamp

**Comment by dpsutton** [comment]
awesome. thanks @alexpusch i'm gonna use that today to keep working on this. thanks for the help

**Comment by dpsutton** [comment]
this was fixed in master and this PR: https://github.com/metabase/metabase/pull/13564 fixes it in the milestone release for 36.8
